### PR TITLE
Ta 2552

### PR DESCRIPTION
**Pull request by JayVirdi**
## What does this pull request do?

Please describe what you did.
Added feature tests to test the case start date error messages around the dates 01/10/2007 and 01/01/1995

## Why make these changes?

Please describe why the changes were needed.
To ensure that we had proper coverage to test incoming changes relating to TA-2556 and others

## Checklist

- [x] Provided link to a JIRA ticket: https://dsdmoj.atlassian.net/browse/TA-2552

### CODE REVIEW

File: features/step_definitions/cwa_outcome_steps.rb [hunk 1]
```diff
@@ -67,6 +66,9 @@
     if !outcome.has_key?("schedule_ref")
             outcome['schedule_ref'] = CWAProvider.submission.schedule_ref
     end
+    if !outcome.has_key?("case_start_date") && !(area_of_law == "Crime Lower")
```
Comment (solomonahmed) [inline features/step_definitions/cwa_outcome_steps.rb:69]: I'm unclear why this addition is needed?

File: features/validations/legal_help/immigration_and_asylum/case_start_date_validation.feature [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+  Scenario Outline: validate case start date for IMMIGRATION
+    When user adds outcomes for "Legal Help" "Immigration" with fields like this:
+      | case_id | schedule_ref | matter_type | stage_reached | case_start_date | procurement_area | access_point |
+      |     211 |              | IMLB:IOUT   | IE            |      31/12/1994 | PA00178          | AP00187      |
+    Then the outcome does not save and gives an error containing:
+      """
+      Case Start Date is before 01-Oct-2007
+      """
+
+  Scenario Outline: validate case start date for HOUSING
```
Comment (solomonahmed) [inline features/validations/legal_help/immigration_and_asylum/case_start_date_validation.feature:102]: The feature file is called immigration_and_asylum but there are housing submissions in it?  Ideally if there is a need for a housing submission test then it needs to be in a legal_help/housing/case_start_date_validation.feature file.

File: run_failed_tests.sh [hunk 3]
```diff
@@ -0,0 +1 @@
+#!/bin/bash
```
Comment (solomonahmed) [inline run_failed_tests.sh:1]: What the purpose of this file?

**Review by solomonahmed — CHANGES_REQUESTED** [review]

File: features/step_definitions/cwa_outcome_steps.rb (hunk 1 above)
Comment (vimalsuresh) [reply features/step_definitions/cwa_outcome_steps.rb:69]: This was added as we had to test error messages in the code 
      Case Concluded Date is before Case Start Date
      Case Start Date is greater than today's date
To do this we had to manipulate the system date which would feed in tomorrows date in the case start date field. giving us this error. To do this we have to not pass "case_start_date" in the examples , but reliased crime lower case tests also do not pass case_start_date. so had to put this code to create exceptions.

File: features/validations/legal_help/immigration_and_asylum/case_start_date_validation.feature (hunk 2 above)
Comment (vimalsuresh) [reply features/validations/legal_help/immigration_and_asylum/case_start_date_validation.feature:102]: I will move it to the respective folder. but the same would have to be done for asylum. Happy to move if , please let me know. I thought case_start_date validations would stay in one place. May me move it out of the categories folders.

File: run_failed_tests.sh (hunk 3 above)
Comment (vimalsuresh) [reply run_failed_tests.sh:1]: This shell file was created to make it easy to run failed tests in the kubernetes cluster. copy pasting the failed tests in to the shell script and running the shell would run the tests one by one.

File: features/step_definitions/cwa_outcome_steps.rb (hunk 1 above)
Comment (solomonahmed) [reply features/step_definitions/cwa_outcome_steps.rb:69]: Need to talk to you about this as I'm not sure about this approach.

File: features/validations/legal_help/immigration_and_asylum/case_start_date_validation.feature (hunk 2 above)
Comment (solomonahmed) [reply features/validations/legal_help/immigration_and_asylum/case_start_date_validation.feature:102]: Let's have a chat.  I take your point re: Immigration And Asylum.  I think the naming conventions need to be consistent and logical.  If there are **Immigration** codes in the this file then they need to be in an **Immigration** folder.  Certainly, Housing tests need to be in a housing directory.

File: features/step_definitions/cwa_outcome_steps.rb (hunk 1 above)
Comment (vimalsuresh) [reply features/step_definitions/cwa_outcome_steps.rb:69]: issue created to look at refactor
https://github.com/ministryofjustice/laa-cwa-feature-tests/issues/202

File: run_failed_tests.sh (hunk 3 above)
Comment (vimalsuresh) [reply run_failed_tests.sh]: the shell has been deleted

File: features/validations/legal_help/immigration_and_asylum/case_start_date_validation.feature (hunk 2 above)
Comment (vimalsuresh) [reply features/validations/legal_help/immigration_and_asylum/case_start_date_validation.feature]: Housing tests are now moved to housing section

**Comment by JayVirdi** [comment]
Test changes made and approved, issue raised for refactor. Merging.

**Pull request merged**